cookbooks/clearwater/recipes/openimscorehss.rb

```diff
@@ -32,53 +32,29 @@
 # under which the OpenSSL Project distributes the OpenSSL toolkit software,
 # as those licenses appear in the file LICENSE-OPENSSL.
 
-%w{subversion mysql-server libmysqlclient-dev libxml2 libxml2-dev bind9 flex bison libcurl4-openssl-dev openjdk-7-jre-headless openjdk-7-jdk ant}.each do |p|
-  package p do
-    action [:install]
-    options "--force-yes"
-  end
-end
-
-directory "/opt/OpenIMSCore" do
-end
-
-subversion "OpenIMSCore HSS" do
-  repository "http://svn.code.sf.net/p/openimscore/code/FHoSS/trunk"
-  destination "/opt/OpenIMSCore/FHoSS"
-  revision "1193"
-  action :sync
+# Tell apt about the repository server.
+cookbook_file "/etc/apt/sources.list.d/fhoss.list" do
+  mode "0644"
+  source "openimscorehss/fhoss.apt.list"
 end
 
-execute "mysql" do
-  command "mysql -uroot --password= </opt/OpenIMSCore/FHoSS/scripts/hss_db.sql && mysql -uroot --password= </opt/OpenIMSCore/FHoSS/scripts/userdata.sql && mysql -uroot --password= </opt/OpenIMSCore/FHoSS/scripts/hss_db_migrate_as_register.sql"
+execute "apt-get update" do
 end
 
-execute "ant" do
-  cwd "/opt/OpenIMSCore/FHoSS"
-  environment ({ "JAVA_TOOL_OPTIONS" => "-Dfile.encoding=UTF-8" })
-  command "ant compile deploy"
+# Install the package, using a response_file to configure it with our IP and home domain
+package "openimscore-fhoss" do
+    action [:install]
+    response_file 'openimscorehss/debian.preseed.erb'
+    options "--force-yes"
 end
 
-domain = if node[:clearwater][:use_subdomain]
-           node.chef_environment + "." + node[:clearwater][:root_domain]
-         else
-           node[:clearwater][:root_domain]
-         end
-
-template "/opt/OpenIMSCore/FHoSS/deploy/DiameterPeerHSS.xml" do
+# Fix up the users so that we just have hssAdmin with a password of the signup key
+template "/usr/share/java/fhoss-0.2/conf/tomcat-users.xml" do
   mode "0644"
-  source "openimscorehss/DiameterPeerHSS.xml.erb"
-  variables domain: domain,
-            node: node
-end
-
-cookbook_file "/etc/init.d/openimscorehss" do
-  mode "0755"
-  source "openimscorehss/openimscorehss.init.d"
-  owner "root"
-  group "root"
+  source "openimscorehss/users.xml"
 end
 
-service "openimscorehss" do
-  action :start
+# Restart to pick up that password change
+service "openimscore-fhoss" do
+  action :restart
```

Comment:
This should be `action :nothing` and be triggered to `:restart` by the previous snippet.